runtime/Cpp/runtime/src/atn/PredictionContext.cpp

```diff
@@ -159,10 +159,10 @@ Ref<PredictionContext> PredictionContext::mergeSingletons(const Ref<SingletonPre
     Ref<PredictionContext> parent = merge(a->parent, b->parent, rootIsWildcard, mergeCache);
 
     // If parent is same object as existing a or b parent or reduced to a parent, return it.
-    if (parent == a->parent) { // ax + bx = ax, if a=b
+    if (*parent == *a->parent) { // ax + bx = ax, if a=b
       return a;
     }
-    if (parent == b->parent) { // ax + bx = bx, if a=b
+    if (*parent == *b->parent) { // ax + bx = bx, if a=b
       return b;
```

Comment:
Now the wrong comparison is back?

Reply:
Sorry, was experimenting with that and left it in.